docs/net.md

```diff
@@ -0,0 +1,343 @@
+# Non-blocking I/O
+
+This section describes the network resources and driver provided by Tokio. This
+component provide one of Tokio's primary functions: non-blocking, event-driven,
+networking provided by the appropriate operating system primitives (epoll,
+kqueue, IOCP, etc...). It is modeled after the resource and driver pattern
+described in the previous section.
+
+The network driver is built using [mio] and network resources are backed by
+types that implement [`Evented`].
+
+This guide will be focused on TCP types. The other network resources (UDP, unix
+sockets, pipes, etc) follow the same pattern.
+
+## The network resource.
+
+Network resources are types, such as [`TcpListener`] and [`TcpStream`] are
+composed of the network handle and a reference to the driver that is powering
+the resource. Initially, when the resource is first created, the driver pointer
+may be `None`:
```

Comment:
The paragraphs below explain the two cases. If an example is provided that shows how to create a listener w/ a handle, is that sufficient?

Reply:
I think that should do it. This wording just confused me a bit when I read it (I did, however, read the docs in the order Github PR review presents them rather than in the way you linked from the first page, which makes things a lot clearer)